ts/src/index.ts

```
@@ -43,60 +42,62 @@ function isHttp2Request(req: RawRequest): req is http2.Http2ServerRequest {
-    if (isHttp2Request(req.req)) {
-      const reqStream = req.req.stream;
-      const cookies = cookie.parse(req.req.headers['cookie'] as string || '');
+  app.register(require('fastify-static'), opts);
+
+  app.addHook('onRequest', async (req, res) => {
+    if (isHttp2Request(req)) {
+      const reqStream = req.stream;
+      const url: string = req.url;
+      let reqPath: string = url.split('?')[0];
+      reqPath = reqPath.replace(prefix, '');
+      (req.stream as StorePath).__req_path = reqPath;
+      const cookies = cookie.parse(req.headers['cookie'] as string || '');
       const cacheKey = cookies[CACHE_COOKIE_KEY];
       const newCacheKey =
           await ap.preprocessRequest(reqPath, reqStream, cacheKey);
       // TODO(jinwoo): Consider making this persistent across sessions.
-      res.header('set-cookie', cookie.serialize(CACHE_COOKIE_KEY, newCacheKey));
-
-      send(req.req, reqPath, {root})
-          .on('error',
-              (err) => {
-                if (err.code === 'ENOENT') {
-                  ap.recordRequestPath(reqStream.session, reqPath, false);
-                  res.code(404).send();
-                } else {
-                  res.code(500).send(err);
-                }
-              })
-          .on('end',
-              () => {
-                ap.recordRequestPath(reqStream.session, reqPath, true);
-              })
-          .pipe(res.res as stream.Writable);
-      await ap.push(reqStream);
-    } else {
-      send(req.req, reqPath, {root})
-          .on('error',
-              (err) => {
-                if (err.code === 'ENOENT') {
-                  res.code(404).send();
-                } else {
-                  res.code(500).send(err);
-                }
-              })
-          .pipe(res.res as stream.Writable);
+      res.setHeader(
+          'set-cookie', cookie.serialize(CACHE_COOKIE_KEY, newCacheKey));
+
+      ap.push(reqStream).then(noop, noop);
```

Any reason you're silencing errors here? Would just `await ap.push(reqStream);` be enough?

---

Two reasons:

1. if push errors (maybe the client rejects that? a file can't be read for some operating system reason?) we should not return a 500 to the user.
2. The processing of the original request can go forward, I do not think there is any reason to wait and postpone the processing here.

---

Makes sense. So does the hook caller awaits until the async hook function finishes?